client/jetpack-connect/utils.js

```diff
@@ -60,7 +60,11 @@ export const authQueryPropTypes = PropTypes.shape( {
 } );
 
 export function addCalypsoEnvQueryArg( url ) {
-	return addQueryArgs( { calypso_env: config( 'env_id' ) }, url );
+	let calypsoEnv = config( 'env_id' );
+	if ( COMMIT_SHA ) {
```

Comment:
`COMMIT_SHA` should be present in almost every build including production. We'll need to adjust this 🙂 

I think we might need to do:

```js
if ( COMMIT_SHA && window.location.host === 'calypso.live' ) { //…
```